<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.9.1"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>Inventaria: numpy.core._exceptions.AxisError Class Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">Inventaria
   </div>
   <div id="projectbrief">A lightweight inventory keeping program</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.9.1 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search','.html');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><b>numpy</b></li><li class="navelem"><a class="el" href="namespacenumpy_1_1core.html">core</a></li><li class="navelem"><a class="el" href="namespacenumpy_1_1core_1_1__exceptions.html">_exceptions</a></li><li class="navelem"><a class="el" href="classnumpy_1_1core_1_1__exceptions_1_1AxisError.html">AxisError</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#pub-methods">Public Member Functions</a> &#124;
<a href="#pub-attribs">Public Attributes</a> &#124;
<a href="classnumpy_1_1core_1_1__exceptions_1_1AxisError-members.html">List of all members</a>  </div>
  <div class="headertitle">
<div class="title">numpy.core._exceptions.AxisError Class Reference</div>  </div>
</div><!--header-->
<div class="contents">
<div class="dynheader">
Inheritance diagram for numpy.core._exceptions.AxisError:</div>
<div class="dyncontent">
<div class="center"><img src="classnumpy_1_1core_1_1__exceptions_1_1AxisError__inherit__graph.png" border="0" usemap="#anumpy_8core_8__exceptions_8AxisError_inherit__map" alt="Inheritance graph"/></div>
<map name="anumpy_8core_8__exceptions_8AxisError_inherit__map" id="anumpy_8core_8__exceptions_8AxisError_inherit__map">
<area shape="rect" title=" " alt="" coords="5,80,216,121"/>
<area shape="rect" title=" " alt="" coords="10,5,99,32"/>
<area shape="rect" title=" " alt="" coords="123,5,213,32"/>
</map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<div class="dynheader">
Collaboration diagram for numpy.core._exceptions.AxisError:</div>
<div class="dyncontent">
<div class="center"><img src="classnumpy_1_1core_1_1__exceptions_1_1AxisError__coll__graph.png" border="0" usemap="#anumpy_8core_8__exceptions_8AxisError_coll__map" alt="Collaboration graph"/></div>
<map name="anumpy_8core_8__exceptions_8AxisError_coll__map" id="anumpy_8core_8__exceptions_8AxisError_coll__map">
<area shape="rect" title=" " alt="" coords="5,80,216,121"/>
<area shape="rect" title=" " alt="" coords="10,5,99,32"/>
<area shape="rect" title=" " alt="" coords="123,5,213,32"/>
</map>
<center><span class="legend">[<a href="graph_legend.html">legend</a>]</span></center></div>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-methods"></a>
Public Member Functions</h2></td></tr>
<tr class="memitem:a463663b3f25704e21f3418c6db30952e"><td class="memItemLeft" align="right" valign="top"><a id="a463663b3f25704e21f3418c6db30952e"></a>
def&#160;</td><td class="memItemRight" valign="bottom"><b>__init__</b> (self, axis, ndim=None, msg_prefix=None)</td></tr>
<tr class="separator:a463663b3f25704e21f3418c6db30952e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aef8b9060b4927783be4810b5a498f08d"><td class="memItemLeft" align="right" valign="top"><a id="aef8b9060b4927783be4810b5a498f08d"></a>
def&#160;</td><td class="memItemRight" valign="bottom"><b>__str__</b> (self)</td></tr>
<tr class="separator:aef8b9060b4927783be4810b5a498f08d"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="pub-attribs"></a>
Public Attributes</h2></td></tr>
<tr class="memitem:ad80038593990d4383fad09cdc79b1b13"><td class="memItemLeft" align="right" valign="top"><a id="ad80038593990d4383fad09cdc79b1b13"></a>
&#160;</td><td class="memItemRight" valign="bottom"><b>axis</b></td></tr>
<tr class="separator:ad80038593990d4383fad09cdc79b1b13"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8568413489d65bf3a7b9516ee18bcaa1"><td class="memItemLeft" align="right" valign="top"><a id="a8568413489d65bf3a7b9516ee18bcaa1"></a>
&#160;</td><td class="memItemRight" valign="bottom"><b>ndim</b></td></tr>
<tr class="separator:a8568413489d65bf3a7b9516ee18bcaa1"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><pre class="fragment">Axis supplied was invalid.

This is raised whenever an ``axis`` parameter is specified that is larger
than the number of array dimensions.
For compatibility with code written against older numpy versions, which
raised a mixture of `ValueError` and `IndexError` for this situation, this
exception subclasses both to ensure that ``except ValueError`` and
``except IndexError`` statements continue to catch `AxisError`.

.. versionadded:: 1.13

Parameters
----------
axis : int or str
    The out of bounds axis or a custom exception message.
    If an axis is provided, then `ndim` should be specified as well.
ndim : int, optional
    The number of array dimensions.
msg_prefix : str, optional
    A prefix for the exception message.

Attributes
----------
axis : int, optional
    The out of bounds axis or ``None`` if a custom exception
    message was provided. This should be the axis as passed by
    the user, before any normalization to resolve negative indices.

    .. versionadded:: 1.22
ndim : int, optional
    The number of array dimensions or ``None`` if a custom exception
    message was provided.

    .. versionadded:: 1.22


Examples
--------
&gt;&gt;&gt; array_1d = np.arange(10)
&gt;&gt;&gt; np.cumsum(array_1d, axis=1)
Traceback (most recent call last):
  ...
numpy.AxisError: axis 1 is out of bounds for array of dimension 1

Negative axes are preserved:

&gt;&gt;&gt; np.cumsum(array_1d, axis=-2)
Traceback (most recent call last):
  ...
numpy.AxisError: axis -2 is out of bounds for array of dimension 1

The class constructor generally takes the axis and arrays'
dimensionality as arguments:

&gt;&gt;&gt; print(np.AxisError(2, 1, msg_prefix='error'))
error: axis 2 is out of bounds for array of dimension 1

Alternatively, a custom exception message can be passed:

&gt;&gt;&gt; print(np.AxisError('Custom error message'))
Custom error message</pre> </div><hr/>The documentation for this class was generated from the following file:<ul>
<li>inventaria_env/lib/python3.10/site-packages/numpy/core/_exceptions.py</li>
</ul>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by&#160;<a href="https://www.doxygen.org/index.html"><img class="footer" src="doxygen.svg" width="104" height="31" alt="doxygen"/></a> 1.9.1
</small></address>
</body>
</html>
